src/main/java/com/amazon/ion/impl/IonReaderContinuableApplicationBinary.java

```diff
@@ -206,6 +214,110 @@ private SymbolTable getSystemSymbolTable() {
+
+        // Note: the first imported table is always the system symbol table.
+        boolean isSystemSymbolTable = numberOfLocalSymbols == 0 && imports.getImportedTablesNoCopy().length == 1;
+        boolean otherHasPrivateAttributes = other instanceof _Private_LocalSymbolTable;
+        _Private_LocalSymbolTable otherLocal = otherHasPrivateAttributes ? (_Private_LocalSymbolTable) other : null;
+        if (isSystemSymbolTable) {
+            if (other.isSystemTable() && maxId == other.getMaxId()) {
+                // Both represent the same system table.
+                lastSupersetSymbolTable = other;
+                return true;
+            }
+            // The other symbol table might not literally be the system symbol table, but if it's a local symbol table
+            // with zero local symbols and zero imports, that counts.
+            if (otherHasPrivateAttributes && otherLocal.getNumberOfLocalSymbols() == 0 && otherLocal.getImportedTablesAsList().isEmpty()) {
+                lastSupersetSymbolTable = other;
+                return true;
+            }
+            return false;
+        }
+        if (!otherHasPrivateAttributes) {
+            // The reader's symbol table is not a system symbol table, but the other is. Other cannot be a superset.
+            return false;
+        }
+        if (maxId > otherLocal.getMaxId()) return false;
+
+        // NOTE: the following uses of _Private_LocalSymbolTable utilize knowledge of the implementation used by
+        // the binary writer, which has the only known use case for this method. Specifically, we call the interface
+        // method variants that return lists instead of arrays because we know this matches the binary writer's symbol
+        // table's internal representation and therefore does not require copying. If this method ends up being used
+        // for other symbol table implementations, which is unlikely, we should add logic to choose the most efficient
+        // variant to call for the particular implementation (such as by adding something like a `boolean usesArrays()`
+        // method to the interface).
+
+        SymbolTable[] readerImports = imports.getImportedTablesNoCopy();
+        if (!compareSymbolTableImportsArrayToList(readerImports, readerImports.length, otherLocal.getImportedTablesAsList())) {
+            return false;
+        }
+
+        // Superset extends subset if subset doesn't have any declared symbols.
+        if (numberOfLocalSymbols == 0) {
+            lastSupersetSymbolTable = other;
+            return true;
+        }
+
+        // Superset must have same/more declared (local) symbols than subset.
+        if (numberOfLocalSymbols > otherLocal.getNumberOfLocalSymbols()) return false;
+
+        Collection<String> otherSymbols = otherLocal.getLocalSymbolsNoCopy();
+        if (!compareSymbolsArrayToCollection(symbols, numberOfLocalSymbols, otherSymbols)) {
+            return false;
+        }
```

Comment:
Now I see that one of the checks I added in my suggestion lives outside the method. Is it correct to push it down? It looks to me like it should be.